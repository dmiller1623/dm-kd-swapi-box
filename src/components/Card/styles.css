.card {
  display: flex;
  flex-direction: column;
  width: 20rem; }

.card-data {
  color: white; }

.clicked {
  background-color: red; }

.fifth-data {
  font-size: 15px;
  display: flex;
  flex-wrap: wrap;
  padding-bottom: 0 auto; }

h5 {
  margin: 0; }
