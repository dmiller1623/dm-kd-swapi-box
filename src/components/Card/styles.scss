.card {
  display: flex;
  flex-direction: column;
  // flex-wrap: wrap;
  width: 20rem;
}

.card-data {

  color: white
}