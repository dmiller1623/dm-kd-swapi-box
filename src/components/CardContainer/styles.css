.card-container {
  width: 100%;
  display: flex;
  height: 50vh; }
