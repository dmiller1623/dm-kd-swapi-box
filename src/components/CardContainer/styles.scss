.card-container {
  width: 100vw;
  display: flex;
  height: 50vh;
}