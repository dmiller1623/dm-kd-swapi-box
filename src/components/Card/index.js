import React from 'react'
import './styles.css'
import PropTypes from 'prop-types'

const Card = (props) => {
  return(
    <div className='card'>
      <h2 className='card-data first-data'>{props.first}
      <div className='button-spacing'>
        <button className={props.evaluateClass(props)} onClick={() => props.updateFavorites(props)}>FAV</button>
      </div>
      </h2>
      <h4 className='card-data'>{props.second}</h4>
      <h4 className='card-data'>{props.third}</h4>
      <h4 className='card-data'>{props.fourth}</h4>
      {props.fifth && <div className='fifth-data'>
        <h5 className='card-data'>Residents:</h5>
        {!props.fifth.length && <h5 className='card-data spacing'>0</h5>}
        { 
          props.fifth.map((person, i) => {
            return <h5 key={i} className='card-data spacing'>{person}</h5>
          })
        }
      </div>}

    </div>
  )
}

Card.propTypes = {
  first: PropTypes.string,
  second: PropTypes.string,
  third: PropTypes.string,
  fourth: PropTypes.string,
  fifth: PropTypes.array
}

export default Card