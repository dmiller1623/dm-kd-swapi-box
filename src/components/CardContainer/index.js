import React from 'react'
import './styles.css'
import Card from '../Card'
import PropTypes from 'prop-types'

const CardContainer = ({ people, vehicles, planets, cardType, favorites, updateFavorites, evaluateClass }) => {
  const peopleData = people.map((person, index) => {
    return <Card 
      {...person}
      key={index}
      updateFavorites={updateFavorites}
      favorites={favorites}
      evaluateClass={evaluateClass}
      />
    })
    
  const planetData = planets.map((planet, index) => {
    return <Card 
      {...planet}
      key={index}
      updateFavorites={updateFavorites}
      favorites={favorites}
      evaluateClass={evaluateClass}
    />
  })
    
  const vehiclesData = vehicles.map((vehicle, index) => {
    return <Card
      {...vehicle}
      favorited={vehicle.favorited}
      key={index}
      updateFavorites={updateFavorites}
      favorites={favorites}
      evaluateClass={evaluateClass}
    />
  })
    
  const favoritesData = favorites.map((favorite, index) => {
    return <Card
      {...favorite}
      key={index}
      favorites={favorites}
      updateFavorites={updateFavorites}
      evaluateClass={evaluateClass}
    />
  }) 
  
  return (
    <div className='card-container'>
      {cardType === 'people' && peopleData}
      {cardType === 'planets' && planetData}
      {cardType === 'vehicles' && vehiclesData}
      {cardType === 'favorites' && 
        (favorites.length ? favoritesData : 
        <div className='favorites-text'>
          No Favorites Selected
        </div>)}
    </div>
  )
}

CardContainer.propTypes = {
  people: PropTypes.array,
  vehicles: PropTypes.array,
  planets: PropTypes.array,
  cardType: PropTypes.string,
  favorites: PropTypes.array,
  updateFavorites: PropTypes.func,
  evaluateClass: PropTypes.func,
}

export default CardContainer