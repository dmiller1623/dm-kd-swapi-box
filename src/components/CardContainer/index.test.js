import React from 'react'
import CardContainer from './index.js'
import { shallow } from 'enzyme'
import Card from '../Card'

describe('CardContainer', () => {
  let people
  let vehicles
  let planets
  let cardType 
  let favorites
  let mockUpdateFavorites
  let mockEvaluateClass 
  let wrapper 
  beforeEach(() => {
    people = []
    vehicles = []
    planets = []
    cardType = "people"
    favorites = []
    mockUpdateFavorites = jest.fn()
    mockEvaluateClass = jest.fn()
    wrapper = shallow(
    <CardContainer 
    people={people}
    vehicles={vehicles}
    planets={planets}
    cardType={cardType}
    favorites={favorites}
    updateFavorites={mockUpdateFavorites}
    evaluateClass={mockEvaluateClass}
    />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render the exact amount of cards nn the people array', () =>{
    people = [ {first: 'first', second: 'second', third: 'third', fourth: 'fourth'} ]
    wrapper = shallow(
      <CardContainer 
      people={people}
      vehicles={vehicles}
      planets={planets}
      cardType={cardType}
      favorites={favorites}
      updateFavorites={mockUpdateFavorites}
      evaluateClass={mockEvaluateClass}
      />)
      expect(wrapper.find(Card).length).toEqual(1)
  })

  it('should render the exact amount of cards in the planets array', () => {
    cardType = 'planets'
    planets = [ {first: 'first', second: 'second', third: 'third', fourth: 'fourth'} ]
    wrapper = shallow(
      <CardContainer 
      people={people}
      vehicles={vehicles}
      planets={planets}
      cardType={cardType}
      favorites={favorites}
      updateFavorites={mockUpdateFavorites}
      evaluateClass={mockEvaluateClass}
      />)
      expect(wrapper.find(Card).length).toEqual(1)
   })  
   
   it('should render the exact amount of cards in the vehicles array', () => {
    cardType = 'vehicles'
    vehicles = [ {first: 'first', second: 'second', third: 'third', fourth: 'fourth'} ]
    wrapper = shallow(
      <CardContainer 
      people={people}
      vehicles={vehicles}
      planets={planets}
      cardType={cardType}
      favorites={favorites}
      updateFavorites={mockUpdateFavorites}
      evaluateClass={mockEvaluateClass}
      />)
      expect(wrapper.find(Card).length).toEqual(1)
   }) 

   it('should render the exact amount of cards in the favorites array', () => {
    cardType = 'favorites'
    favorites = [ {first: 'first', second: 'second', third: 'third', fourth: 'fourth'} ]
    wrapper = shallow(
      <CardContainer 
      people={people}
      vehicles={vehicles}
      planets={planets}
      cardType={cardType}
      favorites={favorites}
      updateFavorites={mockUpdateFavorites}
      evaluateClass={mockEvaluateClass}
      />)
      expect(wrapper.find(Card).length).toEqual(1)
   }) 

   it('should render a second with a message if favorites is empty', () => {
     cardType = 'favorites'
     favorites = []
     wrapper = shallow(
      <CardContainer 
      people={people}
      vehicles={vehicles}
      planets={planets}
      cardType={cardType}
      favorites={favorites}
      updateFavorites={mockUpdateFavorites}
      evaluateClass={mockEvaluateClass}
      />)

      expect(wrapper.find('div').length).toEqual(2)
   })
})