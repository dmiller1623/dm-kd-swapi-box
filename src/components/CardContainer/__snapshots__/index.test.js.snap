// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    cardType="people"
    evaluateClass={[Function]}
    favorites={Array []}
    people={Array []}
    planets={Array []}
    updateFavorites={[Function]}
    vehicles={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [],
        false,
        false,
        false,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      false,
      false,
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [],
          false,
          false,
          false,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        false,
        false,
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
